nmoscommon/utils.py

```diff
@@ -54,3 +78,82 @@ def getLocalIP():
                 pass
     # Could not find an interface
     return None
+
+
+def downgrade_api_version(
+    obj, rtype, target_ver, downgrade_ver=None, downgrade_map=DOWNGRADE_MAP, remove_keys=["max_api_version"]
```

Comment:
Could you set 'downgrade_map' to None by default, then use something like the following in order to avoid the Python anti-pattern:

```
if downgrade_map is None:
    downgrade_map = DOWNGRADE_MAP
```